### PR TITLE
Update documentation-style-guide.md

### PR DESCRIPTION
**Pull request by justinegeffen**
Expanded and added some content, revised tone, included some detail around styles and principles.

### CODE REVIEW

File: operations/operations/company-processes/publishing/publishing-guidelines/voice-tone-and-writing-style-guidelines/documentation-style-guide.md [hunk 1]
```diff
@@ -1,80 +1,123 @@
...
-* For example: "Deployment Guide for Organizations"
+We use Title Case for document names/titles. A document title is usually called an H1 heading and is either denoted by `#` in Markdown or `======` in reStructuredText. The title appears in the left-hand navigation and at the top of the table of contents. Use a title that accurately reflects the content of the document. People scan the table of contents looking for answers; it's often faster than using the built-in search engine.
 
-Use Sentence case for section titles and headings.
+**Use Title Case for document titles \(e.g. "This is an Article about Documentation"\). Where a word/title/name is lowercase, retain that casing in the title \(e.g. "This Article Covers the mmctl Tool"\).**
 
-* For example: "Writing guidelines for editors"
+Instead of *"Deployment guide for organizations"* we write *"Deployment Guide for Organizations"*
+
+#### Section titles and headings
```
Comment (cwarnermm) [inline operations/operations/company-processes/publishing/publishing-guidelines/voice-tone-and-writing-style-guidelines/documentation-style-guide.md:80]: Let's consider covering gerund use here as well for H2 and lower headings in a future iteration of this page. Right now, we're using gerunds inconsistently, and providing clear guidance on our preferences will help contributors going forward.

File: operations/operations/company-processes/publishing/publishing-guidelines/voice-tone-and-writing-style-guidelines/documentation-style-guide.md [hunk 2]
```diff
@@ -309,6 +401,10 @@ The link renders as: [Mattermost Manifesto](https://www.mattermost.org/manifesto
 
 ### Internal links to Mattermost documentation
 
+When creating a link to another document in the Mattermost documentation, create a link with a relative URL.
```
Comment (cwarnermm) [inline operations/operations/company-processes/publishing/publishing-guidelines/voice-tone-and-writing-style-guidelines/documentation-style-guide.md:404]: This is a most excellent point and I'm glad you've included it and provided context as to why a relative link is preferred). At some point in the future, we'll want to clean up all of the absolute links by converting them to relative links.

**Review by cwarnermm — APPROVED** [review]
This Style Guide update is exceptional! Well done! It's clear and easy to read, the thought leadership, guidance, and imperatives include excellent context and explanation, and the examples are easy to digest and learn from. This update very much captures our team's approach to writing instructional content at Mattermost. Thank you so much!!

I've flagged a couple of future considerations inline.

File: operations/operations/company-processes/publishing/publishing-guidelines/voice-tone-and-writing-style-guidelines/documentation-style-guide.md (hunk 1 above)
Comment (justinegeffen) [reply operations/operations/company-processes/publishing/publishing-guidelines/voice-tone-and-writing-style-guidelines/documentation-style-guide.md:80]: Oh that's a great idea, thank you! 

Something like "How to name pages" and then an explanation of gerunds, with examples of avoid and preferred.

**Pull request merged**